src/Interceptor/Impl/CacheInterceptor.php

```diff
@@ -192,16 +276,129 @@ private function getMethodCode(\ReflectionMethod $reflectionMethod): string
     /**
      * @return array<int, string>
      */
-    private function getTags(Instance $instance, Cache $attribute): array
+    private function getTags(Instance $instance, Cache $attribute, mixed $response = null): array
     {
         $parameters = $instance->getMethod()
             ->getParameters();
 
         $tags = array_map(
             static fn (string $expression) => Expression::evaluateToString($expression, $parameters),
-            $attribute->getTags()
+            $attribute->tags
         );
 
-        return array_filter($tags);
+        if ($response !== null) {
+            $tags = array_values(array_filter([
+                ...$tags,
+                ...$this->guessObjectsTags(
+                    $response,
+                    $this->config->autoTagsExcludedClasses
+                ),
+            ]));
+        }
+
+        return $tags;
+    }
+
+    /**
+     * @param array<class-string> $excludedClasses
+     * @param array<string, string> $registeredTags
+     *
+     * @return array<string, string>
+     */
+    private function guessObjectsTags(mixed $object, array $excludedClasses = [], array $registeredTags = []): array
+    {
+        if (!\is_object($object)) {
+            return $registeredTags;
+        }
+
+        foreach ($excludedClasses as $excludedClass) {
+            if ($object instanceof $excludedClass) {
+                return $registeredTags;
+            }
+        }
+
+        $ref = new \ReflectionClass($object);
+
+        $objectTag = $this->buildTag($ref, $object);
+
+        if (isset($registeredTags[$objectTag])) {
+            return $registeredTags;
+        }
+
+        if ($objectTag !== null) {
```

Comment:
we should early return (providing that we'll create a new interface for autotag)